dojo/tools/nuclei/parser.py

```diff
@@ -0,0 +1,75 @@
+import json
+import hashlib
+from dojo.models import Finding, Endpoint
+
+
+class NucleiParser(object):
+    """
+    A class that can be used to parse the nuclei (https://github.com/projectdiscovery/nuclei) JSON report file
+    """
+
+    # table to match nuclei severity to DefectDojo severity
+    SEVERITY = {
+        "critical": "Critical",
+        "high": "High",
+        "medium": "Medium",
+        "low": "Low",
+        "info": "Info",
+    }
+
+    def get_scan_types(self):
+        return ["Nuclei Scan"]
+
+    def get_label_for_scan_types(self, scan_type):
+        return "Nuclei Scan"
+
+    def get_description_for_scan_types(self, scan_type):
+        return "Import JSON output for nuclei scan report."
+
+    def get_findings(self, filename, test):
+        data = [json.loads(line) for line in filename]
+        if len(data) == 0:
+            return []
+        else:
+            dupes = {}
+            for item in data:
+                template_id = item.get('templateID')
+                info = item.get('info')
+                name = info.get('name')
+                if info.get('severity') in self.SEVERITY:
+                    severity = self.SEVERITY[info.get('severity')]
+                else:
+                    severity = "Low"
+                type = item.get('type')
+                host = item.get('host')
+                matched = item.get('matched')
+                ip = item.get('ip')
+
+                dupe_key = hashlib.sha256(
+                    (template_id + type + host + ip).encode('utf-8')
+                ).hexdigest()
+
+                if dupe_key in dupes:
+                    finding = dupes[dupe_key]
+                    finding.nb_occurences += 1
+                else:
+                    finding = Finding(
+                        title=f"{name}",
+                        test=test,
+                        severity=severity,
+                        nb_occurences=1,
+                    )
+                    if info.get('description'):
+                        finding.description = info.get('description')
+                    if info.get('tags'):
+                        finding.unsaved_tags = info.get('tags')
+                    if info.get('reference'):
+                        finding.references = info.get('reference')
+                    finding.unsaved_endpoints = []
+                    if '://' in matched:
```

Comment:
Endpoints shouldn't be added only in "first match" but also when dupe is identified. I suppose move L69-73 should be enough and move L74 after L68

Reply:
It is good to write also unit tests for this - how many endpoints does finding have

Reply:
Yes, you're right. @ansidorov could you add unit tests for endpoints?

Reply:
Fixed